src/standardSchema.ts

```diff
@@ -0,0 +1,176 @@
+
+export interface StandardSchema<Input = unknown, Output = Input> {
+  readonly '~standard': StandardSchemaProps<Input, Output>;
+}
+
+export interface StandardSchemaProps<Input = unknown, Output = Input> {
+  readonly version: 1;
+  readonly vendor: string;
+  readonly validate: (
+    value: unknown,
+  ) => StandardResult<Output> | Promise<StandardResult<Output>>;
+  readonly types?: StandardTypes<Input, Output> | undefined;
+}
+
+type StandardResult<Output> =
+  | StandardSuccessResult<Output>
+  | StandardFailureResult;
+
+interface StandardSuccessResult<Output> {
+  readonly value: Output;
+  readonly issues?: undefined;
+}
+
+interface StandardFailureResult {
+  readonly issues: ReadonlyArray<StandardIssue>;
+}
+
+interface StandardIssue {
+  readonly message: string;
+  readonly path?: ReadonlyArray<PropertyKey | StandardPathSegment> | undefined;
+}
+
+interface StandardPathSegment {
+  readonly key: PropertyKey;
+}
+
+interface StandardTypes<Input, Output> {
+  readonly input: Input;
+  readonly output: Output;
+}
+
+export function createStandardSchemaProps<TIn, Output>(
+  schema: AnySchema,
+): StandardSchemaProps<TIn, Output> {
+  /**
+   * Adapts the schema's validate method to the standard schema's validate method.
+   */
+  async function validate(value: unknown): Promise<StandardResult<Output>> {
+    try {
+      const result = await schema.validate(value, {
+        abortEarly: false,
+      });
+
+      return {
+        value: result as Output,
+      };
+    } catch (err) {
+      if (err instanceof ValidationError) {
+        return {
+          issues: issuesFromValidationError(err),
+        };
+      }
+
+      throw err;
+    }
+  }
+
+  return {
+    version: 1,
+    vendor: 'yup',
+    validate,
+  };
+}
+
+function createStandardPath(path: string | undefined): StandardIssue['path'] {
+  if (!path?.length) {
+    return undefined;
+  }
+
+  // Array to store the final path segments
+  const segments: string[] = [];
+  // Buffer for building the current segment
+  let currentSegment = '';
+  // Track if we're inside square brackets (array/property access)
+  let inBrackets = false;
+  // Track if we're inside quotes (for property names with special chars)
+  let inQuotes = false;
+
+  for (let i = 0; i < path.length; i++) {
+    const char = path[i];
+
+    if (char === '[' && !inQuotes) {
+      // When entering brackets, push any accumulated segment after splitting on dots
+      if (currentSegment) {
+        segments.push(...currentSegment.split('.').filter(Boolean));
+        currentSegment = '';
+      }
+      inBrackets = true;
+      continue;
+    }
+
+    if (char === ']' && !inQuotes) {
+      if (currentSegment) {
+        // Handle numeric indices (e.g. arr[0])
+        if (/^\d+$/.test(currentSegment)) {
+          segments.push(currentSegment);
+        } else {
+          // Handle quoted property names (e.g. obj["foo.bar"])
+          segments.push(currentSegment.replace(/^"|"$/g, ''));
```

Comment:
i don't think yup even handles these cases, i'm not sure it makes sense to do it here, it's misleading, b/c validate will still likely treat these as actual path segments

Reply:
Right, I didn't really encounter this in all my time using `yup` so I didn't consider if it was supported or not. What do you think I should do then?